src/FsCheck/Runner.fs

```diff
@@ -48,34 +48,74 @@ type ParallelRunConfig =
-      QuietOnSuccess: bool
-      ///What to print when new arguments args are generated in test n
-      Every         : int -> list<obj> -> string
-      ///What to print every time a counter-example is successfully shrunk
-      EveryShrink   : list<obj> -> string 
-      ///The Arbitrary instances on this class will be merged in back to front order, i.e. instances for the same generated type at the front
-      ///of the list will override those at the back. The instances on Arb.Default are always known, and are at the back (so they can always be
-      ///overridden)
-      Arbitrary     : list<Type>
-      ///A custom test runner, e.g. to integrate with a test framework like xUnit or NUnit. 
-      Runner        : IRunner
-      ///If set, inputs for property generation and property evaluation will be run in parallel. 
-      ParallelRunConfig : ParallelRunConfig option
-    }
+type Config = private Config of
+                {| ///The maximum number of tests that are run.
+                MaxTest       : int
+                ///The maximum number of tests where values are rejected, e.g. as the result of ==>
+                MaxRejected   : int
+                ///If set, the seed to use to start testing. Allows reproduction of previous runs.
+                Replay        : Replay option
+                ///Name of the test.
+                Name          : string
+                ///The size to use for the first test.
+                StartSize     : int
+                ///The size to use for the last test, when all the tests are passing. The size increases linearly between Start- and EndSize.
+                EndSize       : int
+                ///If set, suppresses the output from the test if the test is successful.
+                QuietOnSuccess: bool
+                ///What to print when new arguments args are generated in test n
+                Every         : int -> list<obj> -> string
+                ///What to print every time a counter-example is successfully shrunk
+                EveryShrink   : list<obj> -> string 
+                ///The Arbitrary instances on this class will be merged in back to front order, i.e. instances for the same generated type at the front
+                ///of the list will override those at the back. The instances on Arb.Default are always known, and are at the back (so they can always be
+                ///overridden)
+                Arbitrary     : list<Type>
+                ///A custom test runner, e.g. to integrate with a test framework like xUnit or NUnit. 
+                Runner        : IRunner
+                ///If set, inputs for property generation and property evaluation will be run in parallel. 
+                ParallelRunConfig : ParallelRunConfig option
+                |} with
+                ///Returns the values for the current Config
+                member this.Values =
+                    let values (Config config) = config
+                    this |> values
```

Comment:
What's the return type of `Values` - is it effectively Tuple?

Reply:
It's an anonymous record. [Anonymous Records](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/anonymous-records).

I was torn between exposing the record, or adding value members:
`member private this.Values =
    let values (Config config) = config
    this |> values
member this.MaxTest = this.Values.MaxTest`

Reply:
Yeah, I'm just not sure if it generates a proper .NET object with properties, or whether it generates a `Tuple` with some metadata for the compiler to hook into. My main concern is that, historically for a record type, adding a new configuration parameter has been a breaking change in the SemVer sense, because effectively that adds a parameter to the ctor of the underlying .NET class F# record types are compiled to. So I'm keen to know whether the actual type (meaning, whatever an anonymous record compiles to) that is being returned here can have new properties added without being a breaking change.

If you don't know this is something that can be dealt with later, but just thought I'd bring it to your attention.

Reply:
It looks like it compiles to a .Net class rather than a tuple. I have updated this so that the whole record is private and the values are exposed via members. Hopefully that solves the issue?